File: TestTaskDNS/Models/Person.cs
using System;

namespace TestTask.Models
{
	[Serializable]
	public class Person
	{
		/// <summary>
		/// Идентификатор сотрудника
		/// </summary>
		public int Id { get; set; }
		
		/// <summary>
		/// Имя сотрудника
		/// </summary>
		public string Name { get; set; }

		/// <summary>
		/// Должность сотрудника
		/// </summary>
		public string Position { get; set; }

		/// <summary>
		/// Дата приема сотрудника
		/// </summary>
		public string EmploymentDate { get; set; }
		
		/// <summary>
		/// Отдел сотрудника
		/// </summary>
		public string Order { get; set; }

		/// <summary>
		/// Непосредственный руководитель сотрудника
		/// </summary>
		public Person Manager { get; set; }

		/// <summary>
		/// Идентификатор руководителя сотрудника
		/// </summary>
		public int ManagerId { get; set; }
	}
}

File: TestTaskDNS/wwwroot/js/Person.ts
class Person {
    id: number;
    name: string;
    employmentDate: string;
    position: string;
    order: string;
    manager: Person;
    managerId: number;
    constructor(id: number = 0, name: string = null, employmentDate: string = null, position: string = null, order: string = null, manager: Person = null, managerId: number = -1) {
        this.id = id;
        this.name = name;
        this.employmentDate = employmentDate;
        this.position = position;
        this.order = order;
        this.manager = manager;
        this.managerId = managerId;
    }
}
